projects/isy-angular-widgets/src/lib/incomplete-date/incomplete-date.component.spec.ts

```diff
@@ -410,23 +227,53 @@ describe('Integration Tests: IncompleteDateComponent', () => {
     expect(errors[errorKey]).toBeDefined();
   });
 
-  it('should return the string "__" correctly', () => {
-    expect(component.transformDatePart('__', 'x')).toBe('xx');
+  const testCases2 = [
+    {inputVal: '__', fillChar: 'x', expected: 'xx'},
+    {inputVal: 'x_', fillChar: 'x', expected: 'xx'},
+    {inputVal: '_x', fillChar: 'x', expected: 'xx'},
+    {inputVal: '1_', fillChar: 'x', expected: '01'},
+    {inputVal: '_1', fillChar: 'x', expected: '01'}
+  ];
+
+  testCases2.forEach(({inputVal, fillChar, expected}) => {
+    it(`should return the string ${inputVal} correctly transformed`, () => {
+      expect(component.transformDatePart(inputVal, fillChar)).toBe(expected);
+    });
   });
 
-  it('should return the string "x_" correctly', () => {
-    expect(component.transformDatePart('x_', 'x')).toBe('xx');
+  it('should return null when transferISO8601 is true', () => {
```

Comment:
Titel klingt so also würde immer null zurückgegeben werden, sobald transferISO8601 = true ist. Das passiert nur, wenn der Input valid und ein unspecified date ist.
Beschreibung entsprechend anpassen.

Reply:
Description and test ha been adjusted